app/misc/binders.scala

```diff
@@ -0,0 +1,36 @@
+package misc
+
+import java.sql.Timestamp
+import java.time.Instant
+import java.time.format.DateTimeFormatter
+import java.time.temporal.TemporalAccessor
+
+import play.api.mvc.QueryStringBindable
+
+/**
+ *  This allows binding custom type params in play routes
+ */
+object binders {
+
+  /**
+   * THis allows to bind to java.sql.Timestamp in play routes
+   */
+  implicit def queryStringBindable(implicit intBinder: QueryStringBindable[String]) = new QueryStringBindable[Timestamp] {
+    override def bind(key: String, params: Map[String, Seq[String]]): Option[Either[String, Timestamp]] = {
+      val dateString: Option[Seq[String]] = params.get(key)
+      try {
+//        TODO: rework this
```

Comment:
What's the current amount of TODOs?)

Reply:
a lot =) It's just a marker for things that we know can be done better, but for _some_ reasons we just roll with easiest solution